bin/pgut/pgut-fe.c

```diff
@@ -62,7 +62,7 @@ setup_workers(int num_workers)
  		if (username && username[0])
  			appendStringInfo(&buf, "user=%s ", username);
  		if (password && password[0])
- 			appendStringInfo(&buf, "password=%s ", password);
+ 			appendStringInfo(&buf, "password='%s' ", password);
```

Comment:
That'll work for most passwords (and mine), but this approach won't work for passwords that contain single quotes.

https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING-KEYWORD-VALUE

To make it bulletproof: when wrapping with single quotes, we'd need to scan the string and:
- Prefix any backslash with a backslash.
- Prefix any single quote with a backslash.

Reply:
That makes sense :+1:. I see couple of ways of resolving other cases:
- The simpler approach is to copy the function [appendConnStrVal()](https://github.com/postgres/postgres/blob/41f18f021a0882eccbeca62e2ed4b66c6b96e9c9/src/fe_utils/string_utils.c#L485-L521). `libpgfeutils` started sharing it only after Postgres 9.6.
- IMHO the proper approach would be to use [PQconnectdbParams()](https://github.com/postgres/postgres/blob/0028b55bc8e2fc80816094620d116e2cbbd8e5f3/src/interfaces/libpq/fe-connect.c#L668-L679), which doesn't require connection parameters values and which is used by most of the PostgreSQL client applications.

Reply:
With the commit https://github.com/reorg/pg_repack/pull/384/commits/135df8b6a6a3b145439315b18539e583cf9fec5c I replaced call of `PQconnectdb()` by `PQconnectdbParams()`. This should be more solid approach to pass password and other connection options, which doesn't require escaping value strings.

Reply:
Much appreciated! I don't see anything glaringly wrong, but I have very little C experience.